test/unit-test/test_lib/test_vis.py

```diff
@@ -73,69 +76,80 @@ def test_generate_probability_distribution(self):
         self.assertEqual(x, -1)
         self.assertEqual(freq, -1)
 
-    @data(
-        (util.load_from_disk(current_dir + "/data/conv_len"), util.load_from_disk(current_dir + "/data/conv_len_fit")))
-    @unpack
-    def test_exponential_curve_fit_and_plot(self, data, expected_result):
-        output = vis.exponential_curve_fit_and_plot(data, current_dir, "exponential_plot_test")
-        # remove the plot created
-        os.remove(current_dir + '/exponential_plot_test.png')
+    @patch("lib.vis.generate_probability_distribution", autospec=True)
+    @patch("lib.util.exponential_curve_func", autospec=True)
+    @patch("lib.vis.curve_fit", autospec=True)
+    def test_exponential_curve_fit_and_plot(self, mock_curve_fit, mock_curve_func, mock_probability_distribution):
+        data = util.load_from_disk(self.test_data_dir + "/vis/conv_len")
+        expected_result = util.load_from_disk(self.test_data_dir + "/vis/conv_len_fit")
+        mock_curve_fit.return_value = util.load_from_disk(self.test_data_dir + "/vis/curve_fit")
+        mock_curve_func.side_effect = self.mock_exponential_curve_func
+        mock_probability_distribution.side_effect = self.mock_generate_probability_distribution
+
+        output = vis.exponential_curve_fit_and_plot(data, self.test_data_dir, "exponential_plot_test")
+        os.remove(self.test_data_dir + '/exponential_plot_test.png')
         assert np.allclose(output, expected_result)
 
-    @data((util.load_from_disk(current_dir + "/data/conv_ref_time"),
-           util.load_from_disk(current_dir + "/data/conv_ref_time_fit_params")))
-    @unpack
-    def test_exponential_curve_fit_and_plot_x_shifted(self, data, expected_result):
-        output = vis.exponential_curve_fit_and_plot_x_shifted(data, current_dir, "exponential_plot_test_shifted")
+    @patch("lib.vis.generate_probability_distribution", autospec=True)
+    @patch("lib.util.exponential_curve_func", autospec=True)
+    @patch("lib.vis.curve_fit", autospec=True)
+    def test_exponential_curve_fit_and_plot_x_shifted(self, mock_curve_fit, mock_curve_func,
+                                                      mock_probability_distribution):
+        data = util.load_from_disk(self.test_data_dir + "/vis/conv_ref_time")
+        expected_result = util.load_from_disk(self.test_data_dir + "/vis/conv_ref_time_fit_params")
+        mock_curve_fit.return_value = util.load_from_disk(self.test_data_dir + "/vis/curve_fit_x_shifted")
+        mock_curve_func.side_effect = self.mock_exponential_curve_func
+        mock_probability_distribution.side_effect = self.mock_generate_probability_distribution
+
+        output = vis.exponential_curve_fit_and_plot_x_shifted(data, self.test_data_dir, "exponential_plot_test_shifted")
         # delete the plot created
-        os.remove(current_dir + '/exponential_plot_test_shifted.png')
+        os.remove(self.test_data_dir + '/exponential_plot_test_shifted.png')
         assert np.allclose(output, expected_result)
```

Comment:
Please verify the mocks.

Reply:
as per our last discussion, we decided not to mock os package api. Similarly no mocking for matplotlib, pyplot, networkx, igraph

Reply:
@bhuvangupta008 I am referring to mock_curve_fit() and mock_curve_func(). It is not about creation of new mocks, but about the verification of mocks already present in the unit test.

Reply:
Yes, it probably that we are using the expected prefix at most places, but I think sometimes the names just becomes too long i. expected_message_number_graph. We can let the expected prefix pattern be for now and decide over it.